instrumentation/opentelemetry-instrumentation-threading/tests/test_threading.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import List
+
+from opentelemetry import trace
+from opentelemetry.instrumentation.threading import ThreadingInstrumentor
+from opentelemetry.test.test_base import TestBase
+
+
+class TestThreading(TestBase):
+    def setUp(self):
+        super().setUp()
+        self._tracer = self.tracer_provider.get_tracer(__name__)
+        self._mock_span_contexts: List[trace.SpanContext] = []
+        ThreadingInstrumentor().instrument()
+
+    def tearDown(self):
+        ThreadingInstrumentor().uninstrument()
+        super().tearDown()
+
+    def get_root_span(self):
+        return self._tracer.start_as_current_span("rootSpan")
+
+    def test_trace_context_propagation_in_thread(self):
+        self.run_threading_test(threading.Thread(target=self.fake_func))
+
+    def test_trace_context_propagation_in_timer(self):
+        self.run_threading_test(
+            threading.Timer(interval=1, function=self.fake_func)
+        )
+
+    def run_threading_test(self, thread: threading.Thread):
+        with self.get_root_span() as span:
+            expected_span_context = span.get_span_context()
+            thread.start()
+            thread.join()
+
+            # check result
+            self.assertEqual(len(self._mock_span_contexts), 1)
+            self.assert_span_context_equality(
+                self._mock_span_contexts[0], expected_span_context
+            )
+
+    def test_trace_context_propagation_in_thread_pool(self):
+        max_workers = 10
+        executor = ThreadPoolExecutor(max_workers=max_workers)
+
+        expected_span_contexts: List[trace.SpanContext] = []
+        futures_list = []
+        for num in range(max_workers):
```

Comment:
To "prove" the other bug is gone, you would want at least one thread in the pool executing at least 2 tasks. The easiest way to guarantee that would be `max_workers=1` and submitting at least 2 tasks.

Reply:
Can I create another test case to do that?

Reply:
Eventually, I decided to create a new test case for thread pool.
- test_trace_context_propagation_in_thread_pool_with_multiple_workers (original)
- test_trace_context_propagation_in_thread_pool_with_single_worker (newer)